lib_comfyui/polling_client.py

```diff
@@ -0,0 +1,164 @@
+        xxx2img = ("img2img" if is_img2img else "txt2img")
+        setattr(global_state, f'{xxx2img}_{workflow_type}_input_images', batch)
+        setattr(global_state, f'tab_name', xxx2img)
+        PromptQueueTracker.update_tracked_id()
+        response = ComfyuiNodeWidgetRequests.send({
+            'request': '/sd-webui-comfyui/webui_request_queue_prompt',
+            'workflowType': f'comfyui_{workflow_type}_{xxx2img}',
+            'requiredNodeTypes': required_node_types,
+            'queueFront': queue_front,
+        })
+
+        if 'error' in response:
+            return response
+
+        # unsafe queue tracking
+        PromptQueueTracker.wait_for_last_put()
+
+        output_key = f'{xxx2img}_{workflow_type}_output_images'
+        results = getattr(global_state, output_key, None)
+
+        if output_key in global_state:
+            delattr(global_state, output_key)
+
+        return results
+
+    @classmethod
+    def set_loop(cls, loop):
+        if loop is None or cls.loop is not None:
+            return
+
+        cls.loop = loop
+        cls._start_listener()
+
+    @classmethod
+    def _start_listener(cls):
+        def request_listener():
+            while request_thread.is_running():
+                try:
+                    cls.last_params = cls.start_comfyui_queue.get(timeout=1)
+                except queue.Empty:
+                    continue
+                key = ComfyuiNodeWidgetRequests.focused_key
+                cls.loop.call_soon_threadsafe(cls.param_events[key][cls.last_params['workflowType']].set)
+
+        def graceful_exit():
+            nonlocal request_thread
+            try:
+                request_listener()
+            except OSError:
+                pass
+
+        request_thread = StoppableThread(target=graceful_exit)
+        request_thread.start()
+        atexit.register(request_thread.join)
+
+    @classmethod
+    def add_client(cls, cid, key):
+        if key not in cls.cids.keys():
+            cls.cids[key] = set()
+        if key not in cls.param_events.keys():
+            cls.param_events[key] = {}
+        if cid in cls.cids[key]:
+            return
+
+        # REMOVE THIS AT SOME POINT
+        ComfyuiNodeWidgetRequests.focused_key = key
+
+        cls.param_events[key][cid] = asyncio.Event()
+        cls.cids[key].add(cid)
+        print(f'[sd-webui-comfyui] registered new ComfyUI client - {cid}')
+
+    @classmethod
+    async def handle_response(cls, response):
+        cls.finished_comfyui_queue.put(response)
+
+    @classmethod
+    async def handle_request(cls, cid, key):
+        await cls.param_events[key][cid].wait()
+        cls.param_events[key][cid].clear()
+        return cls.last_params
+
+
+def polling_server_patch(instance, loop):
+    from aiohttp import web
+
+    ComfyuiNodeWidgetRequests.set_loop(loop)
+
+    @instance.routes.post("/sd-webui-comfyui/webui_polling_server")
+    async def webui_polling_server(request):

```

Comment:
```suggestion
    async def webui_polling_server(response):

```
Just a detail but I think it reads better like this? Feel free to just close this one, I'm not gonna fight for it.